src/main/java/org/jabref/gui/linkedfile/DownloadLinkedFileAction.java

```diff
@@ -164,19 +166,22 @@ private void onSuccess(Path targetDirectory, Path destination) {
             } else {
                 newLinkedFile.setSourceURL(linkedFile.getSourceUrl());
             }
-            entry.replaceDownloadedFile(linkedFile.getLink(), newLinkedFile);
             isHtml = newLinkedFile.getFileType().equals(StandardExternalFileType.URL.getName());
+            if (!isHtml) {
+                entry.replaceDownloadedFile(linkedFile.getLink(), newLinkedFile);
+            }
         }
-
         // Notify in bar when the file type is HTML.
         if (isHtml) {
-            dialogService.notify(Localization.lang("Downloaded website as an HTML file."));
-            LOGGER.debug("Downloaded website {} as an HTML file at {}", linkedFile.getLink(), destination);
+            dialogService.notify(Localization.lang("Download was a HTML file. Not adding it."));
+            List<LinkedFile> newFiles = new ArrayList<>(entry.getFiles());
+            newFiles.remove(linkedFile);
+            entry.setFiles(newFiles);
```

Comment:
I am not sure about this one if there are multiple files attached and only one leads to HTML. Need to see the full source.

Reply:
I'm only removing the current linked File from he current download.
I am getting all  existing files from the entry and just remove the current one
To trigger change detection I have to reassign the list